dlt/_workspace/cli/_runtime_command.py

```diff
@@ -0,0 +1,118 @@
+            logout()
+        else:
+            self.parser.print_usage()
+
+
+def login() -> None:
+    auth_service = AuthService(run_context=active())
+    try:
+        auth_info = auth_service.authenticate()
+        fmt.echo("Already logged in as %s" % fmt.bold(auth_info.email))
+        authorise(auth_service=auth_service)
+    except RuntimeNotAuthenticated:
+        fmt.echo("Logging in with Github OAuth")
+        client = AuthClient(base_url=AUTH_BASE_URL, verify_ssl=False)
+
+        # start device flow
+        login_request = github_oauth_start.sync(client=client)
+        fmt.echo(
+            "Please go to %s and enter the code %s"
+            % (fmt.bold(login_request.verification_uri), fmt.bold(login_request.user_code))
+        )
+        fmt.echo("Waiting for response from Github...")
+
+        while True:
+            time.sleep(login_request.interval)
+            token_response = github_oauth_complete.sync(
+                client=client,
+                body=github_oauth_complete.GithubDeviceFlowLoginRequest(
+                    device_code=login_request.device_code
+                ),
+            )
+            if isinstance(token_response, github_oauth_complete.LoginResponse):
+                auth_info = auth_service.login(token_response.jwt)
+                fmt.echo("Logged in as %s" % fmt.bold(auth_info.email))
+                authorise(auth_service=auth_service)
+                break
+            elif isinstance(token_response, github_oauth_complete.GithubOauthCompleteResponse400):
+                raise RuntimeError("Failed to complete authentication with Github")
+
+
+def logout() -> None:
+    auth_service = AuthService(run_context=active())
+    auth_service.logout()
+    fmt.echo("Logged out")
+
+
+def authorise(auth_service: Optional[AuthService] = None) -> None:
+    if auth_service is None:
+        auth_service = AuthService(run_context=active())
+        auth_service.authenticate()
+
+    try:
+        auth_service.authorise()
+    except LocalWorkspaceIdNotSet:
+        fmt.echo("No workspace id found in local config, using default remote workspace")
+        auth_service.overwrite_local_workspace_id()
+    except WorkspaceIdMismatch as e:
+        fmt.warning(
+            "Workspace id in local config (%s) is not the same as remote workspace id (%s)"
+            % (e.local_workspace_id, e.remote_workspace_id)
+        )
+        should_overwrite = fmt.prompt(
+            "Do you want to overwrite the local workspace id with the remote one?",
+            choices=["yes", "no"],
+            default="yes",
+        )
+        if should_overwrite == "yes":
+            auth_service.overwrite_local_workspace_id()
+            fmt.echo("Local workspace id overwritten with remote workspace id")
+        else:
+            raise RuntimeError("Unable to synchronise remote and local workspaces")
+    fmt.echo("Authorised to workspace %s" % fmt.bold(auth_service.workspace_id))
```

Comment:
Small not important nit-pick. We use us english internally and externally, where authorise is spelt authorize. Ultimately we should probably extract all of those cli strings into a strings file at some point and can get AI to proof-read. That is for later though :)